app/models/kudo.rb

```diff
@@ -53,22 +55,21 @@ def dup?
   end
 
   def cannot_be_author
-    if pseud
-      commentable = nil
-      if commentable_type == "Work"
-       commentable = Work.find_by(id: commentable_id)
-      end
-      if commentable_type == "Chapter"
-       commentable = Chapter.find_by(id: commentable_id).work
-      end
-      kudos_giver = User.find_by(id: pseud.user_id)
-      if commentable.nil?
-        errors.add(:no_commentable,
-                   ts("^What did you want to leave kudos on?"))
-      elsif kudos_giver.is_author_of?(commentable)
-        errors.add(:cannot_be_author,
-                   ts("^You can't leave kudos on your own work."))
-      end
+    return unless user
+
+    commentable = nil
+    if commentable_type == "Work"
+      commentable = Work.find_by(id: commentable_id)
+    elsif commentable_type == "Chapter"
+      commentable = Chapter.find_by(id: commentable_id).work
```

Comment:
Are there any kudos on chapters?

(I don't think this is in the scope of this PR, because the goal as I see it is to keep the changes minimal aside from the switch from pseuds to users. But it might be nice to have an issue to simplify this code in the future.)

Reply:
We have a small number of kudos (100+) on chapters from a bygone era, so this code will have to stay for now.

We can reopen/update https://otwarchive.atlassian.net/browse/AO3-2197 for cleaning up chapter kudos (adding a rake task) and removing chapter-related validations. We should also restrict what types can be used in the polymorphic association, like we did for subscriptions:

https://github.com/otwcode/otwarchive/blob/6f987a7665d309c6f4e749a34787bcab632d864b/app/models/subscription.rb#L11-L15

What do you think?

Reply:
Those all sound like good things to do! Would the plan be to delete the chapter kudos, or transfer them to the works?

Reply:
I think we can try transfering to works, and if duplicate errors happen, deleting them.